File: JyotiKumari/leetcode linked list/Palindrome linked list.py
 Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        if not head or not head.next:
            return True
        
        slow = fast = head
        
        while fast != None and fast.next != None:
            fast = fast.next.next
            slow = slow.next
            
        prev = None
        while slow:
            temp = slow
            slow = slow.next
            temp.next = prev
            prev = temp
            
        while prev:
            if prev.val != head.val:
                return False
            prev = prev.next
            head = head.next
            
        return True

File: Nivedita/Linkedlist/Palindrome Linked list.cpp
class Solution {
public:
    bool isPalindrome(ListNode* head) {
       
        stack<ListNode*>st;
        ListNode*temp=head;
        while(temp!=NULL)
        {
            st.push(temp);
            temp=temp->next;
        }
        temp=head;
        while(temp!=NULL)
        {
            if(temp->val!=st.top()->val)
            {
                return false;
            }
            temp=temp->next;
            st.pop();
        }
        return true;
    }
};
